pkg/controllers/execution/execution_controller.go

```diff
@@ -134,20 +135,43 @@ func (c *Controller) Reconcile(ctx context.Context, req controllerruntime.Reques
 // SetupWithManager creates a controller and register to controller manager.
 func (c *Controller) SetupWithManager(mgr controllerruntime.Manager) error {
 	ctrlBuilder := controllerruntime.NewControllerManagedBy(mgr).Named(ControllerName).
-		WithEventFilter(predicate.GenerationChangedPredicate{}).
```

Comment:
We don't need to remove the general filter, which has already accounted for changes in generation, including changes in deletion timestamps. Therefore, we don't need to alter the filtering for updates; we just need to ignore the deletion events.